driver-core/src/main/com/mongodb/operation/ChangeStreamOperation.java

```diff
@@ -347,13 +340,45 @@ public void onResult(final AsyncBatchCursor<RawBsonDocument> result, final Throw
                 if (t != null) {
                     callback.onResult(null, t);
                 } else {
-                    callback.onResult(new AsyncChangeStreamBatchCursor<T>(ChangeStreamOperation.this,
-                            (AsyncAggregateResponseBatchCursor<RawBsonDocument>) result, binding), null);
+                    AsyncAggregateResponseBatchCursor<RawBsonDocument> cursor = (AsyncAggregateResponseBatchCursor<RawBsonDocument>) result;
+                    callback.onResult(new AsyncChangeStreamBatchCursor<T>(ChangeStreamOperation.this, cursor, binding,
+                            setChangeStreamOptions(cursor.getPostBatchResumeToken(), cursor.getOperationTime())), null);
                 }
             }
         });
     }
 
+    private BsonDocument setChangeStreamOptions(final BsonDocument postBatchResumeToken, final BsonTimestamp operationTime) {
+        BsonDocument resumeToken = null;
+        if (startAfter != null) {
+            resumeToken = startAfter;
+        } else if (resumeAfter != null) {
+            resumeToken = resumeAfter;
+        } else if (startAtOperationTime == null && postBatchResumeToken == null) {
```

Comment:
`MaxWireVersion`?

Reply:
and this: 
>The initial aggregate response had no results.

?

Reply:
I have added the checks for max wire version and empty first batch.